go/cmd/dolt/commands/sql.go

```diff
@@ -418,107 +417,109 @@ func prepend(s string, ss []string) []string {
+			err = prettyPrintResults(ctx, se.db.Root().VRW().Format(), sqlSch, rowIter)
 		}
-		return nil, err
-	case *sqlparser.Insert, *sqlparser.Update:
-		newRoot, sqlSch, rowIter, err := sqlNewEngine(query, root, dEnv)
-		if err == nil {
-			err = prettyPrintResults(ctx, newRoot.VRW().Format(), sqlSch, rowIter)
-		}
-		return newRoot, err
+		return err
 	case *sqlparser.Delete:
-		newRoot, ok := sqlCheckThenDeleteAllRows(ctx, root, s)
+		ok := se.checkThenDeleteAllRows(ctx, s)
 		if ok {
-			return newRoot, nil
+			return nil
 		}
-		newRoot, sqlSch, rowIter, err := sqlNewEngine(query, root, dEnv)
+		sqlSch, rowIter, err := se.query(ctx, query)
 		if err == nil {
-			err = prettyPrintResults(ctx, newRoot.VRW().Format(), sqlSch, rowIter)
+			err = prettyPrintResults(ctx, se.db.Root().VRW().Format(), sqlSch, rowIter)
 		}
-		return newRoot, err
+		return err
 	case *sqlparser.DDL:
 		_, err := sqlparser.ParseStrictDDL(query)
 		if err != nil {
-			return nil, fmt.Errorf("Error parsing DDL: %v.", err.Error())
+			return fmt.Errorf("Error parsing DDL: %v.", err.Error())
 		}
-		return sqlDDL(ctx, dEnv, root, s, query)
+		return se.ddl(ctx, s, query)
 	default:
-		return nil, fmt.Errorf("Unsupported SQL statement: '%v'.", query)
+		return fmt.Errorf("Unsupported SQL statement: '%v'.", query)
 	}
 }
 
 // Processes a single query in batch mode and returns the result. The RootValue may or may not be changed.
-func processBatchQuery(ctx context.Context, query string, dEnv *env.DoltEnv, root *doltdb.RootValue, batcher *dsql.SqlBatcher) (*doltdb.RootValue, error) {
+func processBatchQuery(ctx context.Context, query string, se *sqlEngine, batcher *dsql.SqlBatcher) error {
 	sqlStatement, err := sqlparser.Parse(query)
 	if err != nil {
-		return nil, fmt.Errorf("Error parsing SQL: %v.", err.Error())
+		return fmt.Errorf("Error parsing SQL: %v.", err.Error())
 	}
 
 	switch s := sqlStatement.(type) {
 	case *sqlparser.Insert:
-		return sqlInsertBatch(ctx, dEnv, root, s, batcher)
+		return se.insertBatch(ctx, s, batcher)
 	default:
 		// For any other kind of statement, we need to commit whatever batch edit we've accumulated so far before executing
 		// the query
 		newRoot, err := batcher.Commit(ctx)
 		if err != nil {
-			return nil, err
+			return err
 		}
-		newRoot, err = processQuery(ctx, query, dEnv, newRoot)
+		se.db.SetRoot(newRoot)
+		err = processQuery(ctx, query, se)
 		if err != nil {
-			return nil, err
+			return err
 		}
-		if newRoot != nil {
-			root = newRoot
-			if err := batcher.UpdateRoot(root); err != nil {
-				return nil, err
-			}
+		if err := batcher.UpdateRoot(se.db.Root()); err != nil {
+			return err
 		}
 
-		return root, nil
+		return nil
 	}
 }
 
-// Executes a SQL statement of either SHOW or SELECT and returns values for printing if applicable.
-func sqlNewEngine(query string, root *doltdb.RootValue, dEnv *env.DoltEnv) (*doltdb.RootValue, sql.Schema, sql.RowIter, error) {
+type sqlEngine struct {
```

Comment:
Bunch of overloaded/non-distinguished terms around in here. Open to nomenclature suggestions of course.